DANS-KNAW/dans-datastation-tools
# DD-1066 Prestaging files to ingest large datasets

Fixes DD-1066

# Description of changes
Adds a new command `dv-prestage-files` which will prestage files from a given directory into the Dataverse file root and output the JSON needed to then register those files with the APIs described in https://guides.dataverse.org/en/latest/developers/s3-direct-upload-api.html

# How to test
Create a directory of input files, for example with `generate-random-files.py` from `dans-dev-tools`.  Make sure you run the tool as the `dataverse` user and that `dataverse.files_root` is configured correctly. You can create a new draft dataset or use and existing one. Run the command, e.g. (make sure you save the generated json)

```
/usr/local/bin/dv-prestage-files -d doi:10.5072/FK2/QKZTLV /vagrant/shared/random5/ | jq . > jsondata5.json
```

Then register the new files, e.g.,
```
 curl -X POST -H "X-Dataverse-key: $API_TOKEN" "$SERVER_URL/api/datasets/:persistentId/addFiles?persistentId=$PERSISTENT_IDENTIFIER" -F "jsonData=$JSON_DATA"
```


# Notify

@DANS-KNAW/dataversedans
